st2api/st2api/controllers/v1/action_views.py

```diff
@@ -50,6 +51,18 @@ def _get_action_by_id(id):
             LOG.exception(msg)
             abort(http_client.NOT_FOUND, msg)
 
+    @staticmethod
+    def _get_action_by_ref(ref):
+        try:
+            action_db = Action.get_by_ref(ref)
+            if not action_db:
+                raise ValueError('Referenced action "%s" doesnt exist' % (ref))
+            return action_db
+        except Exception as e:
+            msg = 'Database lookup for ref="%s" resulted in exception. %s' % (ref, e)
```

Comment:
It seems a little odd to raise an exception in this method only to re-catch it 2 lines down. But equally you need to handle generic and the None being returned, so not sure I could think of a cleaner way.